### ISSUE TITLE
nixos-rebuild-ng: git rev parsing for .version-suffix prints unwanted errors

### ISSUE THREAD

**Issue by jlamur** [NixOS/nixpkgs#365222]
## Describe the bug
In the legacy version nixos-rebuild there is a block of code dedicated to writing the current git rev of nixpkgs into the `.version-suffix` file of the source directory. ng took that part of the code into ng but missed the check for the `.git` directory: https://github.com/NixOS/nixpkgs/blob/e61a2da3885cf5846476b643152cedc86516d802/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh#L580

That makes systems that are based on non-git nixpkgs (eg. nix-store based nixpkgs) print some unwanted errors, eg.:
```
$ nixos-rebuild-ng switch -v
debug: nixos_rebuild.process: calling run with args=['nix-instantiate', '--find-file', 'nixpkgs', '-v'], kwargs={'stdout': -1}, extra_env=None
evaluating file '<nix/derivation-internal.nix>'
copying '/nix/store/5wvrlpfrwlmg24ywkybvidyzcvki6bwx-source'...
debug: nixos_rebuild.process: captured output stdout='/nix/store/5wvrlpfrwlmg24ywkybvidyzcvki6bwx-source\n', stderr=None
debug: nixos_rebuild.process: calling run with args=['git', '-C', PosixPath('/nix/store/5wvrlpfrwlmg24ywkybvidyzcvki6bwx-source'), 'rev-parse', '--short', 'HEAD'], kwargs={'stdout': -1}, extra_env=None
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

--- 
The problem should probably be addressed in another discussion, but IMHO, the whole concept of having some tools writting the revision of the nixpkgs git directory is not really the good way to solve the problem here.
It would never work for people (like me) that do not build from channels or a mutable nixpkgs with a .git.

I do not have a better solution to propose, though.
Would it even be better to get rid of the whole thing and stop writting "impurely" the git rev into `.version-suffix` anywhere if there is no complete solution for this problem this way?

Here are some of the tools that do that (I trimmed the output):
```
$ grep -rni '.version-suffix'
pkgs/top-level/make-tarball.nix:32:    echo -n $VERSION_SUFFIX > .version-suffix
pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh:621:            echo -n "$suffix" > "$nixpkgs/.version-suffix" || true
pkgs/by-name/ni/nixos-rebuild-ng/src/nixos_rebuild/__init__.py:323:            (nixpkgs_path / ".version-suffix").write_text(rev)
nixos/modules/installer/cd-dvd/channel.nix:36:        echo -n ${config.system.nixos.versionSuffix} > $out/nixos/.version-suffix
nixos/lib/make-multi-disk-zfs-image.nix:118:      echo -n ${config.system.nixos.versionSuffix} > $out/nixos/.version-suffix
nixos/lib/make-channel.nix:25:    echo -n $VERSION_SUFFIX > .version-suffix
nixos/lib/make-disk-image.nix:317:    echo -n ${config.system.nixos.versionSuffix} > $out/nixos/.version-suffix
nixos/lib/make-single-disk-zfs-image.nix:106:      echo -n ${config.system.nixos.versionSuffix} > $out/nixos/.version-suffix
```

## Notify maintainers

@thiagokokada

**Comment by thiagokokada** [comment]
Instead of checking if the `.git` directory exists, I will probably just capture stderr that will avoid the error being shown.

**Comment by jlamur** [comment]
> Instead of checking if the `.git` directory exists, I will probably just capture stderr that will avoid the error being shown.

Sounds good to me! Maybe you could still show the error in `-v` mode?

**Comment by thiagokokada** [comment]
> > Instead of checking if the `.git` directory exists, I will probably just capture stderr that will avoid the error being shown.
> 
> Sounds good to me! Maybe you could still show the error in `-v` mode?

This would be unnecessary tricky. The output will show in the debug message anyway.

**Comment by thiagokokada** [comment]
Closed by https://github.com/thiagokokada/nixpkgs/commit/37422786020d0b66cb0e308dd34e08544116dd97.

**Comment by thiagokokada** [comment]
Closed by #364698.
